src/VersionManager.cs

```diff
@@ -19,68 +20,162 @@ internal class VersionManager
 
-                string stagingPath = Path.Combine(USER_DR_DIR, "Download.zip");
+            string stagingPath = Path.Combine(USER_DR_DIR, "Download.zip");
 
-                string uri = $"https://www.darkriftnetworking.com/DarkRift2/Releases/{version}/{tier}/{platform}/";
-                if (tier == ServerTier.Pro)
+            string uri = $"https://www.darkriftnetworking.com/DarkRift2/Releases/{version}/{tier}/{platform}/";
+            if (tier == ServerTier.Pro)
+            {
+                string invoiceNumber = GetInvoiceNumber();
+                if (invoiceNumber == null)
                 {
-                    string invoiceNumber = GetInvoiceNumber();
-                    if (invoiceNumber == null)
-                    {
-                        Console.Error.WriteLine(Output.Red($"You must provide an invoice number in order to download Pro DarkRift releases."));
-                        return null;
-                    }
-
-                    uri += $"?invoice={invoiceNumber}";
+                    Console.Error.WriteLine(Output.Red($"You must provide an invoice number in order to download Pro DarkRift releases."));
+                    return false;
                 }
 
-                try
+                uri += $"?invoice={invoiceNumber}";
+            }
+
+            try
+            {
+                using (WebClient myWebClient = new WebClient())
                 {
-                    using (WebClient myWebClient = new WebClient())
-                    {
-                        myWebClient.DownloadFile(uri, stagingPath);
-                    }
+                    myWebClient.DownloadFile(uri, stagingPath);
                 }
-                catch (WebException e)
+            }
+            catch (WebException e)
+            {
+                Console.Error.WriteLine(Output.Red($"Could not download DarkRift {version} - {tier} (.NET {platform}):\n\t{e.Message}"));
+                return false;
+            }
+
+            Console.WriteLine($"Extracting package...");
+
+            Directory.CreateDirectory(fullPath);
+
+            ZipFile.ExtractToDirectory(stagingPath, fullPath, true);
+
+            Console.WriteLine(Output.Green($"Successfully downloaded package."));
+
+            return true;
+        }
+
+        public static bool IsVersionInstalled(Version version, ServerTier tier, ServerPlatform platform)
+        {
+            Version[] versions = GetVersionDirectories(tier, platform);
+            return Array.Exists(versions, (v) => v == version);
+        }
+
+        /// <summary>
+        /// Gets a list of version with specific tier and platform
+        /// </summary>
+        /// <param name="version">The version number required.</param>
+        /// <param name="pro">Whether the pro version should be used.</param>
+        /// <returns>The list of versions or an empty array</returns>
+        public static Version[] GetVersionDirectories(ServerTier tier, ServerPlatform platform)
```

Comment:
I'd probably remove the reference to directories here. While that is how versions are tracked the caller probably doesn't caare about that fact! 🙂